src/pools/selection-strategies.ts

```diff
@@ -54,36 +67,86 @@ class RoundRobinWorkerChoiceStrategy<Worker extends IWorker, Data, Response>
   }
 }
 
+/**
+ * Selects the less recently used worker.
+ *
+ * @template Worker Type of worker which manages the strategy.
+ * @template Data Type of data sent to the worker. This can only be serializable data.
+ * @template Response Type of response of execution. This can only be serializable data.
+ */
+class LessRecentlyUsedWorkerChoiceStrategy<
+  Worker extends IWorker,
+  Data,
+  Response
+> implements IWorkerChoiceStrategy<Worker> {
+  /**
+   * Constructs a worker choice strategy that selects based on less recently used.
+   *
+   * @param pool The pool instance.
+   */
+  public constructor (
+    private readonly pool: IPoolInternal<Worker, Data, Response>
+  ) {}
+
+  /** @inheritdoc */
+  public choose (): Worker {
+    let minNumberOfTasks = Infinity
+    // A worker is always found because it picks the one with fewer tasks
+    let lessRecentlyUsedWorker!: Worker
```

Comment:
This could be only produce an error if `pool.tasks` is empty :thinking:

Reply:
We can add a sanity check on the map length and then throw an error but that should not happen and I'm not in favour of adding more branches to a hot code path until a bug is found. 
I think it's more related to worker pool initialisation timing: a busy loop can be added at init to ensure we wait for at lest a worker to be created.

Reply:
Maybe we should add a check for `numberOfWorkers` to be greater than or equals to `1`
Then in `AbstractPool` constructor the `createAndSetupWorker` function is always called at least once

Reply:
Like for `checkFilePath (filePath: string) `? A 

```ts
checkNumberOfWorkers (numberOfWorkers: number) {
  if (numberOfWorkers === 0) {
    throw Error('')
  }
}
```

method will the call in the constructor? That's ok.